zjit/src/hir.rs

```diff
@@ -1981,8 +1989,11 @@ pub fn iseq_to_hir(iseq: *const rb_iseq_t) -> Result<Function, ParseError> {
     // item between commas in the source increase the parameter count by one,
     // regardless of parameter kind.
     let mut entry_state = FrameState::new(iseq);
-    for idx in 0..num_locals(iseq) {
-        if idx < unsafe { get_iseq_body_param_size(iseq) }.as_usize() {
+    let mut self_param = fun.push_insn(fun.entry_block, Insn::Param { idx: SELF_PARAM_IDX });
```

Comment:
Instead of making this mutable, maybe we should make it loop-local to shrink distance between def and use (might be lower cognitive overhead)